docs/src/manual/nonlinear.md

```diff
@@ -0,0 +1,335 @@
+```@meta
+CurrentModule = JuMP
+DocTestSetup = quote
+    using JuMP
+end
+DocTestFilters = [r"≤|<=", r"≥|>=", r" == | = ", r" ∈ | in ", r"MathOptInterface|MOI"]
+```
+
+# [Nonlinear Modeling](@id new_nonlinear_interface)
+
+!!! warning
+    This page describes an experimental nonlinear interface to JuMP. The API
+    described below is stable, and it will not break with future 1.X releases of
+    JuMP. However, solver support may be limited, and there may be gaps in
+    functionality compared with [Nonlinear Modeling](@ref). To report a bug, or
+    request a missing feature, please [open an issue](https://github.com/jump-dev/JuMP.jl/issues/new/choose).
+
+JuMP has support for general smooth nonlinear (convex and nonconvex)
+optimization problems. JuMP is able to provide exact, sparse second-order
+derivatives to solvers. This information can improve solver accuracy and
+performance.
+
+## Set a nonlinear objective
+
+Use [`@objective`](@ref) to set a nonlinear objective.
+
+```jldoctest
+julia> model = Model();
+
+julia> @variable(model, x[1:2]);
+
+julia> @objective(model, Min, exp(x[1]) - sqrt(x[2]))
+exp(x[1]) - sqrt(x[2])
+```
+
+To modify a nonlinear objective, call [`@objective`](@ref) again.
+
+## Add a nonlinear constraint
+
+Use [`@constraint`](@ref) to add a nonlinear constraint.
+
+```jldoctest nonlinear_constraint
+julia> model = Model();
+
+julia> @variable(model, x[1:2]);
+
+julia> @constraint(model, exp(x[1]) <= 1)
+exp(x[1]) - 1.0 ≤ 0
+
+julia> @constraint(model, con[i = 1:2], 2^x[i] >= i)
+2-element Vector{ConstraintRef{Model, MathOptInterface.ConstraintIndex{MathOptInterface.ScalarNonlinearFunction, MathOptInterface.GreaterThan{Float64}}, ScalarShape}}:
+ con[1] : (2.0 ^ x[1]) - 1.0 ≥ 0
+ con[2] : (2.0 ^ x[2]) - 2.0 ≥ 0
+```
+
+Delete a nonlinear constraint using [`delete`](@ref):
+```jldoctest nonlinear_constraint
+julia> delete(model, con[1])
+```
+
+## Create a nonlinear expression
+
+Use [`@expression`](@ref) to create nonlinear expression objects. The syntax
+is identical to [`@expression`](@ref), except that the expression can contain
```

Comment:
Maybe you need to use the macro to use user defined functions as well ? I would clarify this because the user might be confused when he sees it works outside the macro as well afterwards.

Reply:
I just deleted the sentence